contracts/validators/CredentialAtomicQueryV3Validator.sol

```diff
@@ -79,50 +80,35 @@ contract CredentialAtomicQueryV3Validator is CredentialAtomicQueryValidator {
         return VERSION;
     }
 
-    function parseCommonPubSignals(
-        uint256[] calldata inputs
-    ) public pure override returns (CommonPubSignals memory) {
-        CommonPubSignals memory pubSignals = CommonPubSignals({
-            merklized: inputs[0],
-            userID: inputs[1],
-            circuitQueryHash: inputs[2],
-            requestID: inputs[8],
-            challenge: inputs[9],
-            gistRoot: inputs[10],
-            issuerID: inputs[11],
-            issuerState: inputs[3],
-            isRevocationChecked: inputs[12],
-            issuerClaimNonRevState: inputs[13],
-            timestamp: inputs[14]
-        });
-
-        return pubSignals;
-    }
-
-    function parseV3SpecificPubSignals(
-        uint256[] calldata inputs
-    ) internal pure returns (V3PubSignals memory) {
+    function parsePubSignals(uint256[] calldata inputs) public pure returns (V3PubSignals memory) {
         V3PubSignals memory pubSignals = V3PubSignals({
-            linkID: inputs[4],
-            nullifier: inputs[5],
-            operatorOutput: inputs[6],
-            proofType: inputs[7],
-            verifierID: inputs[15],
-            nullifierSessionID: inputs[16],
-            isBJJAuthEnabled: inputs[17]
+            userID: inputs[0],
+            circuitQueryHash: inputs[1],
+            issuerState: inputs[2],
+            linkID: inputs[3],
+            nullifier: inputs[4],
+            operatorOutput: inputs[5],
+            proofType: inputs[6],
+            requestID: inputs[7],
+            challenge: inputs[8],
+            gistRoot: inputs[9],
+            issuerID: inputs[10],
+            issuerClaimNonRevState: inputs[11],
+            timestamp: inputs[12],
+            isBJJAuthEnabled: inputs[13]
         });
 
         return pubSignals;
     }
 
-    function _verify(
+    function verify(
         uint256[] calldata inputs,
         uint256[2] calldata a,
         uint256[2][2] calldata b,
         uint256[2] calldata c,
         bytes calldata data,
         address sender
-    ) internal view override returns (ICircuitValidator.KeyToInputIndex[] memory) {
+    ) external view override returns (ICircuitValidator.KeyToInputIndex[] memory) {
         CredentialAtomicQueryV3 memory credAtomicQuery = abi.decode(
```

Comment:
should we add circuitId check here ?
` require(
            credAtomicQuery.circuitIds.length == 1 && verifier != IVerifier(address(0)),
            "Invalid circuit ID"
        );`